odoo/addons/base/models/res_users.py

```diff
@@ -1262,6 +1262,18 @@ def _update_user_groups_view(self):
                 E.group(*(xml1), col="2"),
                 E.group(*(xml2), col="2", attrs=str(user_type_attrs)),
                 E.group(*(xml3), col="4", attrs=str(user_type_attrs)), name="groups_id", position="replace")
+
+            #hide groups if all their fields/groups are hidden
+            for group in xml:
```

Comment:
What is the purpose of this code ? not sure to find relevant specs in task ..

Reply:
The goal here is to update the view and hide groups sections if all their child fields/groups are hidden, purpose is to avoid having empty spaces in the form.